systemd/emulation.go

```diff
@@ -153,17 +155,28 @@ func (s *emulation) AddMountUnitFile(snapName, revision, what, where, fstype str
 		return "", fmt.Errorf("cannot mount %s (%s) at %s in preseed mode: %s; %s", what, hostFsType, where, err, string(out))
 	}
 
-	multiUserTargetWantsDir := filepath.Join(dirs.SnapServicesDir, "multi-user.target.wants")
-	if err := os.MkdirAll(multiUserTargetWantsDir, 0755); err != nil {
-		return "", err
+	// In order to handle preseeding of old images with old version of snapd,
+	// we also need to add the unit to multi-user.target
+	targets := []string{
+		"snapd.mounts.target",
+		"multi-user.target",
 	}
 
-	// cannot call systemd, so manually enable the unit by symlinking into multi-user.target.wants
+	// cannot call systemd, so manually enable the unit by symlinking into snapd.mounts.target.wants
```

Comment:
Now that you have more than one target, maybe you can rewrite this to
```suggestion
	// cannot call systemd, so manually enable the unit by symlinking into <target>.wants
```

Reply:
Done